### PR TITLE
Integrate StreamingSplitEstimator code review findings

### PR DESCRIPTION
**Pull request by ahrav**
Addresses findings from code review of gossip-8936.8:

- Stride-based sparse sampling replaces O(N×cap) compaction and per-item heap churn with O(1) amortised observe. Samples are indexed by a dual-axis stride (rank and byte) rather than stored in separate vectors.

- u64 integer math replaces f64 positions to eliminate precision loss above 2^53 bytes. Interpolation uses u128 intermediates to handle byte ranges exceeding 2^63.

- Single `Sample` struct unifies rank and byte checkpoints, eliminating duplicate key storage.

- `Sample` carries both positions and a `sampled_by_bytes` flag for merge phase-alignment; interpolation and selection are now axis-generic.

- Redacted Debug impl for `Sample` to prevent key byte leakage.

- Doc comment on byte-sample logic clarified to match code intent.

- Parameter renamed `compression` → `sample_cap` for clarity.

- Property tests added for stability, byte-precision guards, and performance. Covers Zipf distributions and merge consistency.

Closes gossip-8938, gossip-8939, gossip-8940, gossip-8941, gossip-8942, gossip-8943, gossip-8944.

### CODE REVIEW

File: crates/gossip-connectors/src/split_estimator.rs [hunk 1]
```diff
@@ -1,310 +1,594 @@
...
+        if sample_rank || sample_bytes {
+            // Prefer the byte-stride mark as the recorded position when the
+            // byte trigger fired; this gives tighter byte-space fidelity.
+            let byte_position = if sample_bytes {
+                self.next_byte_mark
+            } else {
+                cumulative_bytes
+            };
+            self.samples
+                .push(Sample::new(rank, byte_position, sample_bytes, key));
```
[the remark this answers is not in the document]
Comment (ahrav) [reply crates/gossip-connectors/src/split_estimator.rs]: Agreed this is a real design trade-off worth documenting. Added explicit notes in two places:

1. Module-level "Design choices" section now has a bullet: **Single byte-mark per `observe` call** — explains the O(1) trade-off, the gap it creates for huge files, and that property tests confirm <1% error on 20k-key Zipf streams.

2. `observe` doc comment now notes: "at most one sample is recorded per call, even if `file_size` spans multiple byte-stride marks."

Not changing the behavior — the O(1) amortised cost is a deliberate choice, and the rank sample for the same file provides fallback coverage. The existing `zipf_like_stream_is_within_one_percent_weight_error` test validates this empirically.

**Comment by ahrav** [comment]
## Response to review findings

### Merge byte-position monotonicity (Critical) — CONFIRMED & FIXED

The dual-offset translation in `merge` (`other_byte_phase` for byte-triggered, `byte_offset` for rank-triggered) does produce non-monotonic byte positions. Wrote `merge_byte_monotonicity_with_close_mixed_triggers` which fails before the fix with a clear inversion (`28544 → 28523`). Fixed by using a uniform `byte_offset` for all samples.

Also added `assert_samples_sorted` after every `merge()` call in the existing tests — good catch that these were missing.

### First-key guard fallback (Logic) — NOT REPRODUCED

Wrote three targeted tests (extreme compaction, merge+compaction, 2-item edge case). All pass. The `selected_sample_indices` algorithm preserves evenly-spaced samples including midpoint coverage, so `nearest_sample(Rank, count/2)` always returns a key past rank 0. See inline reply for details.

### Single byte-mark per `observe` (Logic) — DOCUMENTED

Agreed this is a real trade-off. Added explicit documentation on both `observe` and in the module-level "Design choices" section. Not changing behavior — O(1) per-observe is intentional and property tests confirm <1% error on Zipf streams.

### Summary

| Finding | Verdict | Action |
|---------|---------|--------|
| Merge non-monotonic bytes | Confirmed | Fixed + 4 new tests |
| Missing merge sort assertions | Confirmed | Added to 2 existing tests |
| First-key guard fallback | Not reproduced | 1 new coverage test |
| Single mark per observe | Design trade-off | Documented |

All 24 split_estimator tests pass. `cargo fmt/check/clippy/doc` all clean.
